packages/main/src/CheckBox.ts

```diff
@@ -0,0 +1,464 @@
+import ValueState from "@ui5/webcomponents-base/dist/types/ValueState.js";
+import { getFeature } from "@ui5/webcomponents-base/dist/FeaturesRegistry.js";
+import { getEffectiveAriaLabelText } from "@ui5/webcomponents-base/dist/util/AriaLabelHelper.js";
+import { isSpace, isEnter } from "@ui5/webcomponents-base/dist/Keys.js";
+import "@ui5/webcomponents-icons/dist/accept.js";
+import Icon from "./Icon.js";
+import Label from "./Label.js";
+import WrappingType from "./types/WrappingType.js";
+import {
+	VALUE_STATE_ERROR,
+	VALUE_STATE_WARNING,
+	VALUE_STATE_SUCCESS,
+	// @ts-ignore
+} from "./generated/i18n/i18n-defaults.js";
+
+// Template
+import CheckBoxTemplate from "./generated/templates/CheckBoxTemplate.lit.js";
+
+// Styles
+import checkboxCss from "./generated/themes/CheckBox.css.js";
+import type FormSupport from "./features/InputElementsFormSupport.js";
+import type { IFormElement } from "./features/InputElementsFormSupport.js";
+
+let isGlobalHandlerAttached = false;
+let activeCb: CheckBox;
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * Allows the user to set a binary value, such as true/false or yes/no for an item.
+ * <br><br>
+ * The <code>ui5-checkbox</code> component consists of a box and a label that describes its purpose.
+ * If it's checked, an indicator is displayed inside the box.
+ * To check/uncheck the <code>ui5-checkbox</code>, the user has to click or tap the square
+ * box or its label.
+ * <br><br>
+ * The <code>ui5-checkbox</code> component only has 2 states - checked and unchecked.
+ * Clicking or tapping toggles the <code>ui5-checkbox</code> between checked and unchecked state.
+ *
+ * <h3>Usage</h3>
+ *
+ * You can define the checkbox text with via the <code>text</code> property. If the text exceeds the available width, it is truncated by default.
+ * In case you prefer text to wrap, set the <code>wrappingType</code> property to "Normal".
+ * The touchable area for toggling the <code>ui5-checkbox</code> ends where the text ends.
+ * <br><br>
+ * You can disable the <code>ui5-checkbox</code> by setting the <code>disabled</code> property to
+ * <code>true</code>,
+ * or use the <code>ui5-checkbox</code> in read-only mode by setting the <code>readonly</code>
+ * property to <code>true</code>.
+ *
+ * <br><br>
+ * <h3>Keyboard Handling</h3>
+ *
+ * The user can use the following keyboard shortcuts to toggle the checked state of the <code>ui5-checkbox</code>.
+ * <ul>
+ * <li>[SPACE, ENTER] - Toggles between different states: checked, not checked.</li>
+ * </ul>
+ * <br><br>
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import "@ui5/webcomponents/dist/CheckBox";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webc.main.CheckBox
+ * @extends sap.ui.webc.base.UI5Element
+ * @tagname ui5-checkbox
+ * @public
+ */
+@customElement("ui5-checkbox")
+@languageAware
+/**
+ * Fired when the component checked state changes.
+ *
+ * @public
```

Comment:
we have to link events to JSDoc also. But instead of @name tag we have to use: @event
```js
// @event sap.ui.webc.main.CheckBox#change
```